File: 424.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int i = 0;
        int j = 0;
        int ans = 0;
        int n = s.length();
        unordered_map<char,int> ss;
        int mc = 0;
        while(j < n){
            ss[s[j]]++;
            mc = max(mc, ss[s[j]]);
            if(j-i + 1 - mc > k){
                ss[s[i]]--;
                i++;
            }
            ans = max(ans, j - i + 1);
            j++;
        }
        return ans;
    }
};

File: Python/424.py
class Solution:
    def characterReplacement(self, nums: str, k: int) -> int:
        l = 0
        dd = {}
        sol = 0
        mm = float('-inf')
        for r in range(len(nums)):
            dd[nums[r]] = dd.get(nums[r],0)+1
            mm = max(mm, dd[nums[r]])
            while r - l + 1 - mm > k:
                if dd[nums[l]] == 1:
                    del dd[nums[l]]
                else:
                    dd[nums[l]] -= 1
                l+=1
            sol = max(sol, (r-l+1))
        return sol
